libs/pika/version/src/version.cpp

```diff
@@ -162,73 +101,20 @@ namespace pika {
             full_version_as_string(), PIKA_VERSION_TAG, PIKA_HAVE_GIT_COMMIT);
     }
 
-    std::string boost_version()
-    {
-        // BOOST_VERSION: 107100
-        return pika::util::format("V{}.{}.{}", BOOST_VERSION / 100000,
-            BOOST_VERSION / 100 % 1000, BOOST_VERSION % 100);
-    }
-
-    std::string hwloc_version()
-    {
-        // HWLOC_API_VERSION: 0x00010700
-        return pika::util::format("V{}.{}.{}", HWLOC_API_VERSION / 0x10000,
-            HWLOC_API_VERSION / 0x100 % 0x100, HWLOC_API_VERSION % 0x100);
-    }
-
-#if defined(PIKA_HAVE_MALLOC)
-    std::string malloc_version()
-    {
-        return PIKA_HAVE_MALLOC;
-    }
-#endif
-
-    std::string boost_platform()
-    {
-        return BOOST_PLATFORM;
-    }
-
-    std::string boost_compiler()
-    {
-        return BOOST_COMPILER;
-    }
-
-    std::string boost_stdlib()
-    {
-        return BOOST_STDLIB;
-    }
-
     std::string complete_version()
     {
-        std::string version = pika::util::format("Versions:\n"
+        std::string version = pika::util::format("Version:\n"
```

Comment:
Not for this PR, but I think with fmt/`std::format` we can also make these constexpr. And change the `ostringstream` ones to use `format` as well.